jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1115 @@
+        else:
+            flux = self._jax_children[1]["flux"]
+
+        # If the user specified a flux, then set the flux ratio for the transform that wraps
+        # this class
+        return flux / self._image_flux
+
+    @lazy_property
+    def _image_flux(self):
+        return jnp.sum(self._image.array, dtype=float)
+
+    @lazy_property
+    def _offset(self):
+        # Figure out the offset to apply based on the original image (not the padded one).
+        # We will apply this below in _sbp.
+        offset = self._parse_offset(self._jax_children[1]["offset"])
+        return self._adjust_offset(
+            self._image.bounds, offset, None, self._jax_aux_data["use_true_center"]
+        )
+
+    @lazy_property
+    def _image(self):
+        # Store the image as an attribute and make sure we don't change the original image
+        # in anything we do here.  (e.g. set scale, etc.)
+        if self._jax_aux_data["depixelize"]:
+            # FIXME: no depixelize in jax_galsim
+            # self._image = image.view(dtype=np.float64).depixelize(self._x_interpolant)
+            raise NotImplementedError(
+                "InterpolatedImages do not support 'depixelize' in jax_galsim."
+            )
+        else:
+            image = self._jax_children[0].view(dtype=float)
+
+        if self._jax_aux_data["_recenter_image"]:
+            image.setCenter(0, 0)
+
+        return image
+
+    @lazy_property
+    def _wcs(self):
+        im_cen = (
+            self._jax_children[0].true_center
+            if self._jax_aux_data["use_true_center"]
+            else self._jax_children[0].center
+        )
+
+        # error checking was done on init
+        if self._jax_children[1]["scale"] is not None:
+            wcs = PixelScale(self._jax_children[1]["scale"])
+        elif self._jax_children[1]["wcs"] is not None:
+            wcs = self._jax_children[1]["wcs"]
+        else:
+            wcs = self._jax_children[0].wcs
+
+        return wcs.local(image_pos=im_cen)
+
+    @lazy_property
+    def _jac_arr(self):
+        image = self._jax_children[0]
+        im_cen = (
+            image.true_center if self._jax_aux_data["use_true_center"] else image.center
+        )
+        return self._wcs.jacobian(image_pos=im_cen).getMatrix().ravel()
+
+    @lazy_property
+    def _xim(self):
+        pad_factor = self._jax_aux_data["pad_factor"]
+
+        # The size of the final padded image is the largest of the various size specifications
+        pad_size = max(self._image.array.shape)
+        if pad_factor > 1.0:
+            pad_size = int(math.ceil(pad_factor * pad_size))
+        # And round up to a good fft size
+        pad_size = Image.good_fft_size(pad_size)
+
+        xim = Image(
+            _zeropad_image(
+                self._image.array, (pad_size - max(self._image.array.shape)) // 2
+            ),
+            wcs=PixelScale(1.0),
+        )
+        xim.setCenter(0, 0)
+        xim.wcs = PixelScale(1.0)
```

Comment:
No unfortunately and this caused me some pain. After you call setCenter, you get an OffsetWCS instead of a PixelScale WCS. So you have to reset it. I added a comment.